network/openssl_privatekey.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Yanis Guenane <yanis+ansible@guenane.org>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+from OpenSSL import crypto
+
+import os
+
+DOCUMENTATION = '''
+---
+module: openssl_privatekey
+author: "Yanis Guenane (@Spredzy)"
+version_added: "2.2"
+short_description: Generate OpenSSL private keys
+description:
+    - "This module allows one to (re)generates OepnSSL private keys. It uses
+       the pyOpenSSL python library to interact with openssl. One can generate
+       either RSA or DSA private keys."
```

Comment:
The module do not speak of the format in which the keys would be generated (ie, PEM). It might be worth to add that, since it could be pkcs12, or any other format.